sycl/include/sycl/accessor.hpp

```diff
@@ -1869,6 +1871,12 @@ class __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
   size_t get_count() const { return size(); }
   size_t size() const noexcept { return getAccessRange().size(); }
 
+  size_t byte_size() const noexcept { return size() * sizeof(DataT); }
+
+  size_t max_size() const noexcept { return std::vector<DataT>().max_size(); }
```

Comment:
Out of curiosity, what makes `accessor` have the same `max_size` as `std::vector<DataT>`?

Reply:
Since SYCL2020 describes `max_size()` as `Returns the maximum number of elements any accessor of this type would be able to access.` and the accessor uses `DataT*` as a vector, I thought it should be correct. Do you have another idea?

Reply:
What do you mean "and the accessor uses `DataT*` as a vector"? As noted in [`std::vector<T,Allocator>::max_size`](https://en.cppreference.com/w/cpp/container/vector/max_size) the limit is often just `std::numeric_limits<difference_type>::max()`, but in the end if we rely on `std::vector`'s implementation we make it based on an implementation that I don't believe we use for accessor. Any reason why we couldn't just return `std::numeric_limits<difference_type>::max()` here, apart from `difference_type` not being there yet?

Reply:
> As noted in [`std::vector<T,Allocator>::max_size`](https://en.cppreference.com/w/cpp/container/vector/max_size) the limit is often just `std::numeric_limits<difference_type>::max()`

I understand your concern, we can use numeric_limits here, but as also said in this article:
 "At runtime, the size of the container may be limited to a value smaller than max_size() by the amount of RAM available". So here is a question - which limit should we return in `accessor::max_size()`. I thought it should be the size limited by RAM. I don't mind to change it to opposite if you think so.

Reply:
My reading of the mentioned sentence is that `max_size()` may not actually be the max size at runtime due to physical memory constraints, such as available RAM. However, if that reading is correct that should not affect the value returned by `max_size()`, but it's rather a warning about the the return value not always being the actual limit.

Reply:
Okay, I'll fix it